### PR TITLE
Updated documentation and build process

### PR DESCRIPTION
**Pull request by SorsOps**
- Add documentation site deployment
- Update typing to stop build error
- Prep for more complicated exports
- Convert enums to const enums as reverse  lookups are unnecessary

### CODE REVIEW

**Comment by SorsOps** [comment]
@jorenbroekema

File: tsconfig.json [hunk 1]
```diff
@@ -1,18 +1,26 @@
 {
   "compilerOptions": {
-    "target": "esnext",
```
Comment (jorenbroekema) [inline tsconfig.json:3]: any reason why we want to transpile down to es2017?

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import ts from 'rollup-plugin-ts';
+import pkg from './package.json' assert { type: 'json' };
```
Comment (jorenbroekema) [inline rollup.config.js:2]: I'm not sure about this anymore these days because it will be replaced by:
```
import pkg from './package.json' with { type: 'json' };
```

See: https://github.com/tc39/proposal-import-attributes#history

Turns out, it is now in stage 2 again, for me personally stage 3 is the minimum something needs to be at before I'm willing to use it.. and even though import assertions were in stage 3 and implemented in browsers and node, it was demoted back to stage 2, renamed, and now only has a consensus since last month to go back to stage 3.

Perhaps better to just go with the good old `JSON.parse(fs.readFileSync(filePath, 'utf-8'))` for now? Also saves us from needing the `resolveJsonModule` compiler option in tsconfig I think? Anyway, not a hill for me to die on, so take it or leave it :)

File: tsconfig.json [hunk 3]
```diff
@@ -1,18 +1,26 @@
...
+		"lib": ["es5", "es2015", "es2016", "es2017", "es2018", "es2019", "es2020", "esnext","DOM"],
     "allowSyntheticDefaultImports": true,
     "strict": true,
+    "resolveJsonModule": true,
     "noImplicitThis": true,
     "alwaysStrict": true,
-    "types": ["node"],
+    "types": [
+      "node"
+    ],
```
Comment (jorenbroekema) [inline tsconfig.json:15]: Is this just a random auto-formatting thing that happened? Maybe this repo needs to specify a Prettier config so we can avoid having our personal config differences cause unnecessary formatting diffs?

File: .github/workflows/pages.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# Simple workflow for deploying static content to GitHub Pages
+name: Deploy static content to Pages
+
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (SorsOps) [reply .github/workflows/pages.yml:6]: You're right, it should also be automatic

File: rollup.config.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import ts from 'rollup-plugin-ts';
```
[the remark this answers is not in the document]
Comment (SorsOps) [reply rollup.config.js:1]: It unfortunately doesn't do dts rollup automatically, we'd have to use the extra d.ts rollup plugin as well

File: rollup.config.js (hunk 2 above)
Comment (SorsOps) [reply rollup.config.js:2]: Yeah I didn't want to have to change the strict mode to support the `with` directive and I liked the type safety and intellisense when I used `resolveJsonModule`

File: package.json [hunk 6]
```diff
@@ -1,24 +1,22 @@
...
     "url": "https://github.com/tokens-studio/types.git"
   },
   "type": "module",
-  "exports": {
-    ".": {
-      "types": "./dist/types/index.d.ts"
-    }
-  },
+  "main": "dist/cjs/index.cjs",
+  "module": "dist/esm/index.js",
```
[the remark this answers is not in the document]
Comment (SorsOps) [reply package.json:13]: You're right, I'd prefer to use the latest syntax

File: tsconfig.json (hunk 1 above)
Comment (SorsOps) [reply tsconfig.json:3]: Orignally it was to support the enum outputs in the .js but since moving to const enums this is no longer needed, amending

File: package.json [hunk 7]
```diff
@@ -23,17 +23,18 @@
...
+    "format:prettier": "prettier \"**/*.{ts,js,md,json}\" --write",
     "lint": "run-p lint:*",
     "lint:eslint": "eslint --ext .ts,.html .",
     "lint:prettier": "prettier \"**/*.ts\" --list-different || (echo '↑↑ these files are not prettier formatted ↑↑' && exit 1)",
     "lint:types": "tsc --noEmit",
     "prepare": "husky install",
     "release": "npm run build && changeset publish"
   },
-  "types": "./dist/types/index.d.ts",
+  "types": "dist/esm/index.d.ts",
```
Comment (jorenbroekema) [inline package.json:37]: This change is not correct anymore, still should be ./dist/types/index.d.ts

**Review by jorenbroekema — CHANGES_REQUESTED** [review]

File: package.json (hunk 7 above)
Comment (jorenbroekema) [reply package.json:37]: It  can also be removed entirely btw, similar to main/module field, as it's covered by exports field

**Review by jorenbroekema — APPROVED** [review]
